### ISSUE TITLE
usemin only replaces build:js blocks in certain files

### ISSUE THREAD

**Issue by dudewad** [yeoman/grunt-usemin#554]
There is no real rhyme or reason to this; I'm using handlebars to output the exact same block of JS build blocks into about 40 files.
Indescriminately, it's replacing only about 80% of those blocks, and leaving them untouched in all the remaining files. All of these files are siblings in the same directory.
I'm running htmlmin AFTER I run usemin (due to all the bugs talking about regexp being used to parse html replace blocks), and I've removed a number of other tools to see what the problem is.

Here is my config for useminprepare:

`
    html: "<%= yeoman.app %>/index.html",
    options: {
        dest: '<%= yeoman.dist %>'
    }
`

And my config for usemin:

`
html: ["<%= yeoman.dist %>/**/*.html"],
    options: {
        assetDirs: ["<%= yeoman.dist %>"]
    }
`

As I said, all of these files are being built using the Handlebars generator using the same layout page. They have literally identical head and script block segments.

Line endings are LF for all files (I even tried force converting to other endings and back). 

Any thoughts?

**Comment by eddiemonge** [comment]
Is it the same files failing?

**Comment by dudewad** [comment]
Do you mean every time? Yes, there is a subset of files that fails. I'm trying to see if there's some kind of syntax error that could trigger usemin to fail but I'm not seeing anything at the moment. Been working on this for over 4 hours.

**Comment by dudewad** [comment]
Also, to note, I just updated to the latest version and it had no effect.

**Comment by dudewad** [comment]
Progress: I commented out a component htat was being output by handlebars and it started working.
This has something to do with what Handlebars is outputting. This is strange and seems like a bug in usemin; shouldn't usemin be able to recover gracefully from whatever is on that page, regardless of formatting, etc?

**Comment by dudewad** [comment]
Does bad output include things like, closing html caret on it's own line? As far as I know these things are legal but maybe disliked by regex. Do you know how far along they've come with the html parser version? I hear there's just a grunt-htmparser plugin you can use (snicker snicker).
All jokes aside, I would like to know when the parser will be implemented. Unfortunately I'm neck deep in a lot of code that's utilizing this and I don't have the option to back it out.

**Comment by dudewad** [comment]
As a feature request, I think it'd be smart to add a failure message/warning at the least to tell you where it bailed out of parsing files.

**Comment by eddiemonge** [comment]
Itd probably be faster to fix your code to not output bad stuff then wait on this to get the html parser

**Comment by dudewad** [comment]
I'm not getting errors, thats the thing. It's valid code :/
Still narrowing it down. I'll let you know when I figure out what exactly is causing the issue.

**Comment by jyoung913** [comment]
dudewad I had a very similar problem as yours, usemin task won't replace blocks with generated html files, while useminPrepare, concat, cssmin and uglify all worked properly. 
Replacing all line endings to CRLF solved my problem but I think there's a defect in usemin task, apparently blocks are detected but somehow replacements are missing because of some line ending inconsistency I assume.

Windows 8.1 x64  grunt-contrib-usemin 3.0.0

**Comment by dudewad** [comment]
@jyoung913 you're right!! Thank you! CRLF is the required line ending, at least on windows (would it be different on mac/unix?)

The problem is that my template files are CRLF, but the assemble/handlebars output was LF only, something I can't find a workaround for.

**So, per a suggestion in another ticket, I added the grunt-lineending task to my project, and before running usemin I converted all line endings to CRLF explicitly.**

@eddiemonge this is a less than desirable workaround, since it involves "fixing" code that shouldn't care. It's valid HTML. I think this underscores the urgency to get rid of regex and get a real html parser in place! Love the tool but line-ending stuff makes me worried about working on distributed, cross-platform development environments.
